src/Memoize.jl

```diff
@@ -12,6 +12,12 @@ macro memoize(args...)
         error("Memoize accepts at most two arguments")
     end
 
+    rettype = :nothin
```

Comment:
Would that say that any function that does not declare a return type is return type any?
Is that the correct semantics of return types? It makes sense that it would be.
That is are the functions:
```julia
function f(x)::Any
  return x
end
```
and 
```julia
function f(x)
  return x
end
```
(Semantically) Equivalent? The syntax as seen by macros is definitely different because of the additional `::` node in the AST.